### ISSUE TITLE
Better presets management

### ISSUE THREAD

**Issue by woj-tek** [AwesomeTTS/AwesomeTTS#91]
Hi, given recent changed to the availability of services and having TTS run based on deck name, it would be awesome if it was possible to change current preset setting - right now it requires reentering name of all the decks which may be a bit of a pain.

If there were possible to select one of the presets, selects another service and then simply hit save (and then it would update the preset configuration) it would make the live soooo much easier.

Or possibly allow single preset to be available under a couple of names - "en" preset with configured "english deck one" and "english deck two" and selected some service. Then management would require either changing the service for single preset or adding/removing deck names.

**Comment by corpulentcoffee** [comment]
If I recall correctly, the way we got you setup was basically to use
"{{Deck}}"/"{{Subdeck}}" in your card templates?... And then you created
presets with those names?

You can actually overwrite the settings for a preset today, but you must
manually type its name in.

One bit that makes this tricky is the default for a preset name is "service
(voice)" and I don't want people to accidentally to overwrite "Service A
(Alice)" with Bob from Service B.

I guess we could improve things in one of two ways...

- after selecting a preset, if you change its settings, and its name is not
of the "sss (vvv)" form, and then you hit save, prepopulate the preset name
input text with the original name; or
- whenever you save a preset, the text field should actually be a select
box + text field that allows you to either select an old non-"sss (vvv)"
preset name or type a new one (I need to make sure Qt 4  has support for a
widget like this).

Either of these probably needs a warning that you are about to overwrite a
preset.

Would either of these improvements work for you?

**Comment by woj-tek** [comment]
Yes, that's what I use in the templates. And yes - I know I can overwrite the settings by providing the same name, but either typing them in or copying before (TTS settings window is modal!) is a bit of pain.

I think that the second approach would be better (i.e. single field which allows either selecting existing one or typing in another one).

**Comment by corpulentcoffee** [comment]
Great, I'll look into that. Thanks!
